src/library/Uno.Toolkit.Material/Styles/Controls/v2/TabBar.xaml

```diff
@@ -1,888 +1,1414 @@
+        BasedOn="{StaticResource BaseMaterialTabBarStyle}"
+        TargetType="utu:TabBar">
+        <Setter Property="Background" Value="{ThemeResource MaterialTopTabBarBackground}" />
+        <Setter Property="VerticalAlignment" Value="Top" />
+        <Setter Property="MinHeight" Value="{StaticResource MaterialTopTabBarHeight}" />
+        <Setter Property="utu:SafeArea.Insets" Value="Top" />
+        <Setter Property="ItemContainerStyle" Value="{StaticResource MaterialTopTabBarItemStyle}" />
+        <Setter Property="SelectionIndicatorContentTemplate">
+            <Setter.Value>
+                <DataTemplate>
+                    <Border
+                        Height="2"
+                        Margin="0,0,0,1"
+                        VerticalAlignment="Bottom"
+                        Background="{ThemeResource MaterialTopTabBarBottomBorderBrush}" />
+                </DataTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+
+    <Style
+        x:Key="MaterialColoredTopTabBarStyle"
+        BasedOn="{StaticResource MaterialTopTabBarStyle}"
+        TargetType="utu:TabBar">
+        <Setter Property="Background" Value="{ThemeResource MaterialColoredTopTabBarBackground}" />
+        <Setter Property="ItemContainerStyle" Value="{StaticResource MaterialColoredTopTabBarItemStyle}" />
+        <Setter Property="SelectionIndicatorContentTemplate">
+            <Setter.Value>
+                <DataTemplate>
+                    <Border
+                        Height="2"
+                        Margin="0,0,0,1"
+                        VerticalAlignment="Bottom"
+                        Background="{ThemeResource MaterialColoredTopTabBarBottomBorderBrush}" />
+                </DataTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+    <!--#endregion-->
+
+    <!--#region TabBarItem Styles-->
+    <!--
+        "Navigation" Base TabBarItem Style defines commonalities between:
+        - VerticalTabBar (known as Navigation Rail in Material Design 3: https://m3.material.io/components/navigation-rail/overview) and,
+        - BottomTabBar (known as NavigationBar in Material Design 3: https://m3.material.io/components/navigation-bar/overview)
+    -->
+    <Style x:Key="MaterialBaseNavigationTabBarItemStyle" TargetType="utu:TabBarItem">
+        <Setter Property="FontFamily" Value="{ThemeResource MaterialMediumFontFamily}" />
+        <Setter Property="FontSize" Value="{ThemeResource LabelMediumFontSize}" />
+        <Setter Property="FontWeight" Value="{ThemeResource LabelMediumFontWeight}" />
+        <Setter Property="UseSystemFocusVisuals" Value="True" />
+        <Setter Property="HorizontalContentAlignment" Value="Center" />
+        <Setter Property="Background" Value="{ThemeResource TabBarItemBackground}" />
+        <Setter Property="Foreground" Value="{ThemeResource TabBarItemForeground}" />
+        <Setter Property="BorderBrush" Value="{ThemeResource TabBarItemBorderBrush}" />
```

Comment:
Ugh, the TabBar styles are such a mess right now. Let's avoid using the base TabBarItemBackground/Foreground/BorderBrush resource keys and create a MaterialNavigationTabBarBackground/Foreground/BorderBrush + VisualState variants instead. Same way you did it for the NavigationTabBarItem resources

In the long run, we should remove the "TopTabBarStyle" and it should just be "TabBarStyle" and it could then use the resource keys from the base style. For now let's just use Material specific brushes :/